File: backend/src/main/java/com/teammanager/service/ProjectService.java
package com.teammanager.service;

import com.teammanager.dto.ProjectRequest;
import com.teammanager.dto.ProjectResponse;
import com.teammanager.dto.UserResponse;

import java.util.List;

public interface ProjectService {
    ProjectResponse createProject(ProjectRequest request, String currentUserEmail);
    List<ProjectResponse> getAllProjects(String currentUserEmail);
    ProjectResponse getProjectById(Long id, String currentUserEmail);
    ProjectResponse updateProject(Long id, ProjectRequest request);
    void deleteProject(Long id);
    void addMemberToProject(Long projectId, Long userId);
    List<UserResponse> getProjectMembers(Long projectId);
}

File: frontend/src/services/projectService.js
import api from './api';

const projectService = {
  getAllProjects: async () => {
    const response = await api.get('/projects');
    return response.data.data;
  },
  
  getProjectById: async (id) => {
    const response = await api.get(`/projects/${id}`);
    return response.data.data;
  },
  
  createProject: async (projectData) => {
    const response = await api.post('/projects', projectData);
    return response.data.data;
  },
  
  updateProject: async (id, projectData) => {
    const response = await api.put(`/projects/${id}`, projectData);
    return response.data.data;
  },
  
  deleteProject: async (id) => {
    const response = await api.delete(`/projects/${id}`);
    return response.data;
  },

  addMemberToProject: async (projectId, userId) => {
    const response = await api.post(`/projects/${projectId}/members/${userId}`);
    return response.data;
  },

  getProjectMembers: async (projectId) => {
    const response = await api.get(`/projects/${projectId}/members`);
    return response.data.data;
  }
};

export default projectService;
